src/Microsoft.Data/MutableDataFrameBuffer.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A basic mutable store to hold values in a DataFrame column. Supports wrapping with an ArrowBuffer
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal class MutableDataFrameBuffer<T> : DataFrameBuffer<T>
```

Comment:
Why is DataFrameBuffer public but this one is internal? Do either need to be public?

Reply:
I left the read only version public for now to write unit tests. Once Arrow's `StringArray.Builder.Append` can accept nulls(I'll file a bug separately for that), I'll change both the buffers to internal

Reply:
I almost wonder if it would be better to use reflection or `InternalsVisibleTo` instead. That way we have the right public API.

Reply:
Oh yeah I'm good with that. I thought using `InternalsVisibleTo` was frowned upon? But I guess this is temporary